### PR TITLE
Infra255 fixing minor issues with local setup

### PR DESCRIPTION
**Pull request by mixmixmix**
## Description

When the [Azure infrastructure Ansible work was added](https://github.com/fjelltopp/fjelltopp-ansible/commit/563b0a09403b659c620e7752ce36f720d7ee7c6d), we saw a few bugs introduced to AWS / Local Ansible work.   

This PR fixes some of those bugs in order to get WRC local and staging working.  These bugs include:

- Missing `storage_class_name` Ansible variable default
- `POSTGRES_PASSWORD` env missing from the ckandb_job.yml task. 
- Different volume names expected by Azure and AWS/local (@cooper667 To fix this in a future PR, and to remove the temporary `fjelltopp_cloud_provider` variable and if statement)

Relates to PR https://github.com/fjelltopp/fjelltopp-infrastructure/issues/255

## Checklist

Put an `x` in the boxes that apply to this pull request (you can also fill these out after opening the pull request).
You may not need to check all boxes.

- [ ] The Jira ticket for this issue has been updated to "Ready to Review" or equivalent.
- [ ] I have developed these changes in discussion with the appropriate project manager.
- [ ] My code follows the general Fjelltopp documentation (see Confluence).
- [ ] I have made corresponding changes to the Fjelltopp documentation (see Confluence).
- [ ] I have rebased this branch with master.
- [ ] New dependency changes have been committed.
- [ ] I have added automated tests that prove my fix is effective or that my feature works.
- [ ] New and existing tests pass locally with my changes.
- [ ] My changes generate no new warnings.
- [ ] I have performed a self-review of my own code.
- [ ] I have assigned at least one reviewer.
- [ ] I have assigned at least one label to this PR: "patch", "minor", "major".

### CODE REVIEW

**Comment by cooper667** [comment]
As far as I can see `storage_class_name` _should_ only be needed if env != local?

However we should add `storageClassName: "efs-client"` as a default because AWS uses that and we override it in the Azure yaml where needed

File: roles/ckan/templates/kubernetes/ckandb_job.yaml [hunk 1]
```diff
@@ -24,7 +24,9 @@ spec:
               value: "{{ ckan_db_hostname }}"
             - name: POSTGRES_USER
               value: "{{ rds_admin_username }}"
-            
+            - name: POSTGRES_PASSWORD
+              value: "{{ ckan_postgres_password }}"
+
```
Comment (jonathansberry) [inline roles/ckan/templates/kubernetes/ckandb_job.yaml:29]: Don't understand why this is needed?

**Review by jonathansberry — CHANGES_REQUESTED** [review]
I think I'm going to be a bit anal and call you out on not using the agreed PR template.  There's a lot about this PR that I don't understand, and it's probably because it's missing the context (as requested in the template). Namely - I'm after an explanation of the problem and how you think you have fixed it.

**Comment by mixmixmix** [comment]
> I think I'm going to be a bit anal and call you out on not using the agreed PR template. There's a lot about this PR that I don't understand, and it's probably because it's missing the context (as requested in the template). Namely - I'm after an explanation of the problem and how you think you have fixed it.

Sorry, could have been more verbose. Basically, all those changes are re-introducing variables that are needed by local deployment and got removed by this commit 563b0a09403b659c620e7752ce36f720d7ee7c6d

File: roles/ckan/templates/kubernetes/ckandb_job.yaml (hunk 1 above)
Comment (mixmixmix) [reply roles/ckan/templates/kubernetes/ckandb_job.yaml:29]: postgresdb pod fails with authentication error as it requires password on local deployment, which got accidentally removed 
[d7a722bb3e5cfde6179d410f9d287152fc8c09d4290b675a7aac57369fb55a2dR28](https://github.com/fjelltopp/fjelltopp-ansible/pull/1/files#diff-d7a722bb3e5cfde6179d410f9d287152fc8c09d4290b675a7aac57369fb55a2d)

File: roles/ckan/templates/kubernetes/ckan.yaml [hunk 2]
```diff
@@ -225,6 +228,17 @@ spec:
         - name: ckan-resources
           persistentVolumeClaim:
             claimName: ckan-resources-blob
+{% else %}
```
Comment (mixmixmix) [inline roles/ckan/templates/kubernetes/ckan.yaml:227]: @cooper667 could you help me understand why the claim name is different between azurea nad local (and what about AWS?) ckan-storage-blob vs ckan-storage ? Is this if-else here sensible way to solve that?

**Comment by mixmixmix** [comment]
> As far as I can see `storage_class_name` _should_ only be needed if env != local?
> 
> However we should add `storageClassName: "efs-client"` as a default because AWS uses that and we override it in the Azure yaml where needed

I see. Will set this variable in all.

File: roles/ckan/templates/kubernetes/ckan.yaml (hunk 2 above)
Comment (cooper667) [reply roles/ckan/templates/kubernetes/ckan.yaml:227]: It should both be the same. will merge when we actually have a staging env and remove the difference

File: roles/ckan/templates/kubernetes/ckan.yaml [hunk 3]
```diff
@@ -149,8 +149,10 @@ spec:
               value: "{{ ckan_postgres_password }}"
             - name: PIPENV_DONT_LOAD_ENV
               value: "1"
+{% if fjelltopp_cloud_provider == 'azure' %}
             - name: POSTGRES_HOST
               value: "{{ db_fqdn }}"
+{% endif %}
```
Comment (jonathansberry) [inline roles/ckan/templates/kubernetes/ckan.yaml]:
```suggestion
```
Advice from @cooper667 is just to delete these lines.  This env var shouldn't be needed.

File: group_vars/all/all.yml [hunk 4]
```diff
@@ -29,6 +29,7 @@ jenkins_aws_account_id: "791438942407"
 # Environment type - local is for Minikube, aws for AWS :)
 # by default it's local
 fjelltopp_env_type: "local"
+fjelltopp_cloud_provider: ''
```
Comment (jonathansberry) [inline group_vars/all/all.yml:32]: We want to minimise the number of if statements as far as possible, so adding in another switch variable should be avoided as far as possible.  In this case I think it is acceptable as a temporary measure until Craig resolves the problem of the volume naming.

**Review by jonathansberry — CHANGES_REQUESTED** [review]
I've added the PR template myself and given some suggested context.  It's taken me a little while to figure this out myself so would be helpful if you could add something similar next time.  There's one other change requested based on a chat with Craig.

**Comment by mixmixmix** [comment]
TBH, the file diff says pretty much the same as the PR description... ;) 
And the reason for those changes is really just the fact that variables were missing/added in the breaking PR so there is not much to go by apart form just testing.

**Review by jonathansberry — APPROVED** [review]
> TBH, the file diff says pretty much the same as the PR description... ;) And the reason for those changes is really just the fact that variables were missing/added in the breaking PR so there is not much to go by apart form just testing.

It's just been quite hard to do the cross check of "what you believe you are doing" and "what you are actually doing" and then "why are you doing it" is important.  And even though it's brief, it provides some context and orientation to the PR. Doesn't need to be long, but something is far preferable to nothing, and helps massively with the review process.

**Pull request merged**